### DISCUSSION TITLE
Showing a Fragment from a NativeModule

### DISCUSSION THREAD

**Discussion by 39otrebla** [software-mansion/react-native-screens#1456] [Q&A]
We have the following use case:

- a third-party SDK provides us with a method to build a Fragment to consume their service
- we need to show this Fragment from a NativeModule

We tried the following:

```java
ReactActivity activity = (ReactActivity) mReactContext.getCurrentActivity();

if (activity != null) {
  FragmentManager fragmentManager = activity.getSupportFragmentManager();
  Fragment fragment = MySDK.getFragment(..........);

  if (fragment == null) {
    promise.reject("fragment_error", "Could not get fragment from SDK");
    return;
  }
  
  fragmentManager.beginTransaction().show(fragment).commit();
  // also did an attempt with fragmentManager.beginTransaction().add(fragment, "my_fragment_tag").commit();
  promise.resolve(true);
}
```

What happens is that the Fragment gets instantiated (we see the logs of some API calls happening upon mount), but the Fragment is not visible (neither by our eyes nor by Layout Inspector). It's definitively there (we analyzed the `getFragments()` list from `FragmentManager`), but not showing up.

We also did try this:

- taken the `ScreenStackFragment` from `getFragments()`
- got the SupportFragmentManager from ScreenStackFragment's `activity`
- ran the transaction from there

Same result.

Is there any known way to do this? If possible, I'd also love to be able to just inject our custom Fragment in a react-native-screens Screen, in that way I could potentially reuse TopBar options.

**Comment by Mati882023** [comment]
> We have the following use case:
> 
> - a third-party SDK provides us with a method to build a Fragment to consume their service
> - we need to show this Fragment from a NativeModule
> 
> We tried the following:
> 
> ```java
> ReactActivity activity = (ReactActivity) mReactContext.getCurrentActivity();
> 
> if (activity != null) {
>   FragmentManager fragmentManager = activity.getSupportFragmentManager();
>   Fragment fragment = MySDK.getFragment(..........);
> 
>   if (fragment == null) {
>     promise.reject("fragment_error", "Could not get fragment from SDK");
>     return;
>   }
>   
>   fragmentManager.beginTransaction().show(fragment).commit();
>   // also did an attempt with fragmentManager.beginTransaction().add(fragment, "my_fragment_tag").commit();
>   promise.resolve(true);
> }
> ```
> 
> What happens is that the Fragment gets instantiated (we see the logs of some API calls happening upon mount), but the Fragment is not visible (neither by our eyes nor by Layout Inspector). It's definitively there (we analyzed the `getFragments()` list from `FragmentManager`), but not showing up.
> 
> We also did try this:
> 
> - taken the `ScreenStackFragment` from `getFragments()`
> - got the SupportFragmentManager from ScreenStackFragment's `activity`
> - ran the transaction from there
> 
> Same result.
> 
> Is there any known way to do this? If possible, I'd also love to be able to just inject our custom Fragment in a react-native-screens Screen, in that way I could potentially reuse TopBar options.
>
